### PR TITLE
[NOMERGE ]ffmpeg: update to 4.3.

### PR DESCRIPTION
**Pull request by jnbr**

### CODE REVIEW

**Comment by ericonr** [comment]
Seems like there was a bug in chromium because of the release.

https://bugs.archlinux.org/task/67020

**Comment by jnbr** [comment]
thank you for the heads up

**Comment by q66** [comment]
We should also throw in a revert of https://github.com/FFmpeg/FFmpeg/commit/3a557c5d88b7b15b5954ba2743febb055549b536 but only conditionally, for `ppc64*` but not `ppc64le*`

**Comment by HadetTheUndying** [comment]
Can confirm Chromium bugs are not Arch specific.

**Comment by jnbr** [comment]
> We should also throw in a revert of FFmpeg/FFmpeg@3a557c5 but only conditionally, for ppc* but not ppc64le* (the new code is valid for POWER8 VSX which LE targets as a baseline but not BE)

Are the different baselines specific to void or is it defined by the ABI? I.e. is there a chance to have a single universal patch? I really don't like having conditional patches.

**Comment by q66** [comment]
no, it's universal/broken on all distros with BE now. The conditional revert is the simplest to do in this case (least amount of work/code), proper fix should go in upstream (it's being tracked in freebsd as well as in gentoo, so i expect it to get fixed)

**Comment by jnbr** [comment]
Gentoo now has a patch for chromium: https://gitweb.gentoo.org/repo/gentoo.git/tree/www-client/chromium/files/chromium-84-mediaalloc.patch?id=fd3c6eeae58085ad2248e99ad67d8eba0f3a25ff

**Comment by ericonr** [comment]
`4.3.1` is out already.

**Comment by jnbr** [comment]
Updated to 4.3.1 and added the altivec patch.
With chromium being patched, I think this is good to go.

File: srcpkgs/ffmpeg/template [hunk 1]
```diff
@@ -33,6 +33,22 @@ case "$XBPS_TARGET_MACHINE" in
...
 esac
 
+_apply_patch() {
+	local args="$1" pname="$(basename $2)"
+
+	if [ ! -f ".${pname}_done" ]; then
+		patch -N $args -i $2
+		touch .${pname}_done
+	fi
+}
```
Comment (ericonr) [inline srcpkgs/ffmpeg/template:43]: Could we make this an exported function to be used in the case of conditional patches?

Comment (jnbr) [reply srcpkgs/ffmpeg/template:43]: We could but I think conditional patches are a terrible idea and we should not make it more convenient to add them.

Comment (ericonr) [reply srcpkgs/ffmpeg/template:43]: Fair enough.

**Comment by q66** [comment]
I'm not home till monday so i will look then

**Comment by ericonr** [comment]
Maybe we can merge this one with #23542 ?

**Comment by jnbr** [comment]
merged

**Pull request closed without merging**